mitmproxy/addons/cut.py

```diff
@@ -88,26 +88,29 @@ def save(
         if path.startswith("+"):
             append = True
             path = mitmproxy.types.Path(path[1:])
-        if len(cuts) == 1 and len(flows) == 1:
-            with open(path, "ab" if append else "wb") as fp:
-                if fp.tell() > 0:
-                    # We're appending to a file that already exists and has content
-                    fp.write(b"\n")
-                v = extract(cuts[0], flows[0])
-                if isinstance(v, bytes):
-                    fp.write(v)
-                else:
-                    fp.write(v.encode("utf8"))
-            ctx.log.alert("Saved single cut.")
-        else:
-            with open(path, "a" if append else "w", newline='', encoding="utf8") as fp:
-                writer = csv.writer(fp)
-                for f in flows:
-                    vals = [extract(c, f) for c in cuts]
-                    writer.writerow(
-                        [strutils.always_str(x) or "" for x in vals]  # type: ignore
-                    )
-            ctx.log.alert("Saved %s cuts over %d flows as CSV." % (len(cuts), len(flows)))
+        try:
+            if len(cuts) == 1 and len(flows) == 1:
+                with open(path, "ab" if append else "wb") as fp:
+                    if fp.tell() > 0:
+                        # We're appending to a file that already exists and has content
+                        fp.write(b"\n")
+                    v = extract(cuts[0], flows[0])
+                    if isinstance(v, bytes):
+                        fp.write(v)
+                    else:
+                        fp.write(v.encode("utf8"))
+                ctx.log.alert("Saved single cut.")
+            else:
+                with open(path, "a" if append else "w", newline='', encoding="utf8") as fp:
+                    writer = csv.writer(fp)
+                    for f in flows:
+                        vals = [extract(c, f) for c in cuts]
+                        writer.writerow(
+                            [strutils.always_str(x) or "" for x in vals]  # type: ignore
+                        )
+                ctx.log.alert("Saved %s cuts over %d flows as CSV." % (len(cuts), len(flows)))
+        except (PermissionError, IsADirectoryError, FileNotFoundError) as e:
```

Comment:
FileNotFound does not make sense here, but how about we just say `except IOError`?

Reply:
@mhils I think `FileNotFound` makes sense :)
We can get this exception, when inputting `/ome/file.txt` as a filepath. 
I was thinking about `OSError`, but `IOError` looks more reasonable, thanks.